extendr-api/src/ownership.rs

```diff
@@ -154,7 +158,7 @@ impl Ownership {
     // Garbage collect the tracking structures.
     unsafe fn garbage_collect(&mut self) {
         // println!("garbage_collect {} {}", self.cur_index, self.max_index);
-        let new_size = self.cur_index + EXTRA_PRESERVATION_SIZE;
+        let new_size = self.cur_index * 2 + EXTRA_PRESERVATION_SIZE;
```

Comment:
Why was this chaned? It is always difficult to chose best initial size and growth pattern for growable collections. I'd be very careful with such changes unless we benchmark memory usage in a typical extend scenario. 
Using default values, the first growth will happen from 10k to 30k (if I am not mistaken), the second will jump from 30k to 70k. I am not sure it is justified.